runtime/src/deterministic.rs

```diff
@@ -1408,39 +1446,34 @@ mod tests {
             cycle: Duration::default(),
             ..Config::default()
         };
-        Executor::init(cfg);
+        deterministic::Runner::new(cfg);
     }
 
     #[test]
     fn test_recover_synced_storage_persists() {
         // Initialize the first runtime
-        let (executor1, context1, auditor1) = Executor::default();
+        let executor1 = deterministic::Runner::default();
         let partition = "test_partition";
         let name = b"test_blob";
-        let data = b"Hello, world!".to_vec();
-
-        // Run some tasks and sync storage
-        executor1.start({
-            let context = context1.clone();
-            let data = data.clone();
-            async move {
-                let blob = context.open(partition, name).await.unwrap();
-                blob.write_at(&data, 0).await.unwrap();
-                blob.sync().await.unwrap();
-            }
+        let data = b"Hello, world!";
+
+        // Run some tasks, sync storage, and recover the runtime
+        let (state, context) = executor1.start(|context| async move {
```

Comment:
nit: suggest returning `context` first and giving `state` a name related to `audit` (as `context` actually has application state)